test/test-api-pass-error.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+
+// Testcase for passing an error object to the API call.
+
+if (process.argv[2] === 'child') {
+  const nodereport = require('../');
+  try {
+    throw new Error("Testing error handling");
+  } catch (err) {
+    nodereport.triggerReport(err);
+  }
+} else {
+  const common = require('./common.js');
+  const spawn = require('child_process').spawn;
+  const tap = require('tap');
+
+  const child = spawn(process.execPath, [__filename, 'child']);
+  child.on('exit', (code) => {
+    tap.plan(3);
+    tap.equal(code, 0, 'Process exited cleanly');
+    const reports = common.findReports(child.pid);
+    tap.equal(reports.length, 1, 'Found reports ' + reports);
+    const report = reports[0];
+    common.validate(tap, report, {pid: child.pid,
+      commandline: child.spawnargs.join(' '),
```

Comment:
One of the problems we had with `process.argv` (https://github.com/nodejs/node-report/pull/58#discussion_r101174204) is that Node.js rewrites argv (e.g. it resolves `argv[0]`). When we spawn we have control over the command line, so we know that if we spawn with a resolved path (e.g. `process.execPath`) the operating system will match what is reported by Node.js. 

But as I pointed out elsewhere the command line check is optional -- We can leave out `commandline` from the options passed to `common.validate` and the check will be skipped.